packages/google-vertex/src/google-vertex-image-model.ts

```diff
@@ -0,0 +1,84 @@
+import { ImageModelV1, JSONValue } from '@ai-sdk/provider';
+import { Resolvable, resolve } from '@ai-sdk/provider-utils';
+
+export type GoogleVertexImageModelId =
+  | 'imagen-3.0-generate-001'
+  | 'imagen-3.0-fast-generate-001';
+
+interface GoogleVertexImageModelOptions {
+  provider: string;
+  baseURL: string;
+  headers?: Resolvable<Record<string, string | undefined>>;
+  fetch?: typeof fetch;
+}
```

Comment:
wonder if we can extract generic model options for google (across different kinds of models)

Reply:
I think we can and it's a good idea but would like to do it as a separate change -- will note for follow-up.